pkg/solana/txm/txm.go

```diff
@@ -820,32 +793,124 @@ func (txm *Txm) Enqueue(ctx context.Context, accountID string, tx *solanaGo.Tran
 	return nil
 }
 
-func (txm *Txm) waitForTxStatus(ctx context.Context, transactionID string, desiredStatus commontypes.TransactionStatus) (commontypes.TransactionStatus, error) {
+func (txm *Txm) waitForDependencyTxs(msg pendingTx) {
+	ctx, cancel := txm.chStop.NewCtx() // NOTE: waitForTxStatus will merge this with TxConfirmTimeout
+	defer cancel()
+	depMeta := msg.cfg.DependencyTxMeta
+	err := txm.waitForTxStatus(ctx, depMeta)
+	if err != nil {
+		// IgnoreDependencyError is used by clean up transactions that are expected to be dropped in normal scenarios
+		// No need to log or store error for dependent transactions if dependency tx reached unexpected status
+		if depMeta.IgnoreDependencyError {
+			return
+		}
+		txm.lggr.Errorw("dependency transactions did not reach desired statuses", "id", msg.id, "error", err)
+		err = txm.txs.OnPrebroadcastError(msg.id, txm.cfg.TxRetentionTimeout(), txmutils.Errored, TxDependencyFail)
+		if err != nil {
+			txm.lggr.Errorw("failed to mark transaction as errored", "id", msg.id, "error", err)
+		}
+		return
+	}
+
+	client, err := txm.client.Get(ctx)
+	if err != nil {
+		txm.lggr.Errorw("failed to get client while waiting for dependency transactions", "error", err)
+		return
+	}
+	// Fetch latest blockhash
+	blockhash, err := client.LatestBlockhash(ctx)
+	if err != nil {
+		txm.lggr.Errorw("failed to fetch latest blockhash ", "error", err)
+		return
+	}
+	// Update the dependent transaction's blockhash because waiting for dependency transactions reduced the existing one's validity
+	msg.tx.Message.RecentBlockhash = blockhash.Value.Blockhash
+	msg.lastValidBlockHeight = blockhash.Value.LastValidBlockHeight
+
+	// Estimate compute unit limit if flag is enabled and set the appropriate value in msg config before queueing
+	msg, err = txm.setMsgComputeUnitLimit(ctx, msg)
+	if err != nil {
+		return
+	}
+
+	select {
+	case txm.chSend <- msg:
+		txm.lggr.Debugw("enqueued tx after dependencies reached desired status", "id", msg.id, "dependencyTxCount", len(depMeta.DependencyTxs))
+	default:
+		txm.lggr.Errorw("failed to enqueue tx after dependencies", "queueFull", len(txm.chSend) == MaxQueueLen, "tx", msg)
+	}
+}
+
+func (txm *Txm) waitForTxStatus(ctx context.Context, depMeta txmutils.DependencyTxMeta) error {
```

Comment:
probably a name change makes sense
```suggestion
func (txm *Txm) waitForDependencyTxs(ctx context.Context, depMeta txmutils.DependencyTxMeta) error {
```